resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilder.scala

```diff
@@ -48,7 +54,7 @@ private[spark] class KubernetesDriverBuilder {
       new HadoopConfDriverFeatureStep(conf),
       new KerberosConfDriverFeatureStep(conf),
       new PodTemplateConfigMapStep(conf),
-      new LocalDirsFeatureStep(conf))
+      new LocalDirsFeatureStep(conf)) ++ userFeatures
```

Comment:
what happens on userFeatures failing or throwing?  Do we print a decent log message?

Reply:
If it fails/throws we'll simply pass that exception up which I think is the right approach.

Reply:
ok, I guess my point was does that print something useful and do we have a test for it?  its one thing for our internal code, once folks can start extending and doing potentially weird things I just want to make sure  we have decent message there.

Reply:
So you'll get a stack trace because we don't wrap anything so the developer should see their customs step involved and know that it's up to them to debug it.